aspnet/mvc/controllers/filters.rst

```diff
@@ -1,8 +1,327 @@
+
+.. literalinclude:: filters/sample/src/FiltersSample/Filters/SampleLoggingActionFilterAttribute.cs
+  :language: c#
+  :emphasize-lines: 1, 3, 7
+  :lines: 8-38
+  :dedent: 4
+
+This filter can be applied to classes or methods using the ``[SampleLoggingActionFilter]`` syntax.
+
+``IFilterFactory`` implements ``IFilter``. Therefore, an ``IFilterFactory`` instance can be used as an ``IFilter`` instance anywhere in the filter pipeline. When the framework prepares to invoke the filter, attempts to cast it to an ``IFilterFactory``. If that cast succeeds, the ``CreateInstance`` method is called to create the ``IFilter`` instance that will be invoked. This provides a very flexible design, since the precise filter pipeline does not need to be set explicitly when the application starts.
+
+You can implement ``IFilterFactory`` on your own attribute implementations as another approach to creating filters:
+
+.. literalinclude:: filters/sample/src/FiltersSample/Filters/AddHeaderWithFactoryAttribute.cs
+  :language: c#
+  :emphasize-lines: 1,4-7
+  :lines: 6-26
+  :dedent: 4
+
+.. _ordering:
+
+Ordering
+^^^^^^^^
+Filters can be applied to action methods or controllers (via attribute) or added to the global filters collection. Scope also generally determines ordering. The filter closest to the action runs first; generally you get overriding behavior without having to explicitly set ordering. This is sometimes referred to as "Russian doll" nesting, as each increase in scope is wrapped around the previous scope, like a `nesting doll <https://en.wikipedia.org/wiki/Matryoshka_doll>`_.
+
+In addition to scope, filters can override their sequence of execution by implementing `IOrderedFilter <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNet/Mvc/Filters/IOrderedFilter/index.html>`_. This interface simply exposes an ``int`` ``Order`` property, and filters execute in ascending numeric order based on this property. All of the built-in filters, including ``TypeFilterAttribute`` and ``ServiceFilterAttribute``, implement ``IOrderedFilter``, so you can specify the order of filters when you apply the attribute to a class or method. By default, the ``Order`` property is 0 for all of the built-in filters, so scope is used as a tie-breaker and (unless ``Order`` is set to a non-zero value) is the determining factor.
+
+To see the scope-based ordering in action, consider the following class, which has the same filter applied at both the class and method level. The same filter is also registered globally. When run, the log output will show the order in which the filters executed, and how these compared to the ``Controller`` base class's ``OnActionExecuting`` and ``OnActionExecuted`` methods.
+
+.. literalinclude:: filters/sample/src/FiltersSample/Controllers/OrderingController.cs
```

Comment:
My mistake - I thought filter execution was already logged by MVC. I've opened https://github.com/aspnet/Mvc/issues/4438 to consider adding this to the framework. Regardless, we want to keep the focus of logging on application level concerns instead of framework mechanics. 

You can still document what the filter ordering will be in a specific example, but we don't want to demonstrate the order using logging. Just write what the execution order will be in prose. You can still show logging as an example of DI in filters, but the data logged should specific to the application or the semantics of the filter.